SpeechResponder/CustomFunctions/ShipCallsign.cs

```diff
@@ -17,36 +20,87 @@ public class ShipCallsign : ICustomFunction
         public string description => Properties.CustomFunctions_Untranslated.ShipCallsign;
         public NativeFunction function => new NativeFunction((values) =>
         {
-            int? localId = (values.Count == 0 ? (int?)null : (int)values[0].AsNumber);
+            // The game provides three options for callsigns used by in-game ATC:
+            // (1) CommanderName (default): ship manufacturer name (sometimes partial) + first 3 alphanumeric characters of commander name
+            // (2) ShipName: ship manufacturer name (sometimes partial) + first 3 alphanumeric characters of ship name
+            // (3) ShipID: ship manufacturer name (sometimes partial) + first 3 alphanumeric characters of ship identifier
+
+            int? localId = (values.Count < 0 && values[0].Type != ValueContent.Void 
+                ? (int)values[0].AsNumber 
+                : (int?)null);
+            int? callsignType = (values.Count > 1 && values[1].Type != ValueContent.Void 
+                ? (int)values[1].AsNumber 
+                : (int?)null);
+
             ShipMonitor shipMonitor = (ShipMonitor)EDDI.Instance.ObtainMonitor("Ship monitor");
-            Ship ship = shipMonitor.GetShip(localId);
+            Ship ship = localId is null ? shipMonitor.GetCurrentShip() : shipMonitor.GetShip(localId);
+
+            string result = "";
```

Comment:
Consider a StringBuilder maybe? But only once the tests are in place ;)
  https://docs.microsoft.com/en-us/dotnet/api/system.text.stringbuilder?view=net-5.0

Reply:
Done.